reporting/src/main/java/org/wso2/carbon/testgrid/reporting/reader/ResultReaderFactory.java

```diff
@@ -47,6 +50,11 @@ public static ResultReadable getResultReader(Path path) throws ReportingExceptio
         // Initialize result reader based on the file's extension.
         if (path.toString().endsWith(CSV_EXTENSION)) {
```

Comment:
Shall we add a switch statement instead of if?

Reply:
Since we are performing an operation (in the if statement), it is not possible to convert this to a switch statement directly. If we are to convert to a switch statement, we need to do a string manipulation, which will be more complicated and slow.